core/src/main/java/org/wildfly/extras/creaper/core/online/HttpModelControllerClient.java

```diff
@@ -1,108 +1,108 @@
 
 /**
  * Controller provides execution of {@link ModelNode} or {@link Operation} over HTTP.
  * Asynchronous execution is not provided. Controller does not inform about any progress, therefore, there is no point
  * in using {@link OperationMessageHandler}.
- * If an error occurs (server does not respond 401 or header does not contain WWW-Authenticate field after
- * first request) {@link IllegalStateException} is thrown.
  * Execution on {@link Operation} is allowed with <b>no attachments</b>.
  */
 class HttpModelControllerClient implements ModelControllerClient {
     private static final int NO_TIMEOUT = 0;
 
-    private final CredentialsProvider credsProvider;
-    private final HttpHost target;
     private final String url;
     private final RequestConfig requestConfig;
-    private final boolean areCredentialsGiven;
-    private CloseableHttpClient httpClient;
-    private HttpClientContext localContext;
-
-    public HttpModelControllerClient(String host, int port, String username, String password) {
-        this(host, port, username, password, NO_TIMEOUT);
-    }
-
-    public HttpModelControllerClient(String host, int port, String username, String password, int timeoutMillis) {
-        target = new HttpHost(host, port, "http");
-        credsProvider = new BasicCredentialsProvider();
-        if (username != null && password != null) {
-            credsProvider.setCredentials(
-                    new AuthScope(target.getHostName(), target.getPort()),
-                    new UsernamePasswordCredentials(username, password));
-            areCredentialsGiven = true;
-        } else {
-            areCredentialsGiven = false;
-        }
+    private final String host;
+    private final int port;
+    private final String username;
+    private final String password;
+    private final String managementRealm;
+    private final Registry<ConnectionSocketFactory> registry;
+    private final SSLContext sslContext;
+    private final SSLConnectionSocketFactory sslConnectionSocketFactory;
+
+    private HttpModelControllerClient(
+            String scheme, String host, int port, String username, String password, int timeoutMillis, SslOptions ssl) {
+        this.host = host;
+        this.port = port;
+        this.username = username;
+        this.password = password;
+        url = String.format("%s://%s:%s/management", scheme, host, port);
         RequestConfig.Builder configBuilder = RequestConfig.copy(RequestConfig.DEFAULT);
         if (timeoutMillis != NO_TIMEOUT) {
             configBuilder
                     .setConnectTimeout(timeoutMillis)
                     .setSocketTimeout(timeoutMillis);
         }
+        if (ssl != null) {
+            sslContext = ssl.createSslContext();
+        } else {
+            sslContext = org.apache.http.ssl.SSLContexts.createDefault();
+        }
+        sslConnectionSocketFactory = new SSLConnectionSocketFactory(sslContext, NoopHostnameVerifier.INSTANCE);
```

Comment:
Is there something wrong with using that?

I mean, I agree that this should probably be configurable, but I'd like the HTTPS support in default configuration to be as reasonable as possible and also as close as possible to the native SSL protocol.